### PR TITLE
feat: impl `zks_getTransactionDetails`

### PR DESCRIPTION
**Pull request by grw-ms**
# What :computer: 
* Added `get_transaction_details` to `http_fork_choice`
* Implemented handler for `zks_getTransactionDetails` to zks namespace

# Why :hand:
* Allow users to retrieve tx details

# Evidence :camera:
image
image

# Notes :memo:
* We are not caching these responses due to their mutable status. We could cache when `status == Finalized`, but didn't seem worth it. wdyt?
* Fixes https://github.com/matter-labs/era-test-node/issues/146

### CODE REVIEW

File: e2e-tests/test/zks-apis.test.ts [hunk 1]
```diff
@@ -57,3 +59,18 @@ describe("zks_getTokenPrice", function () {
     expect(response).to.equal("1500");
   });
 });
+
+describe("zks_getTransactionDetails", function () {
+  it.only("Should return transaction details for locally-executed transactions", async function () {
```
Comment (MexicanAce) [inline e2e-tests/test/zks-apis.test.ts:64]: Remove `.only` please
```suggestion
  it("Should return transaction details for locally-executed transactions", async function () {
```

File: src/http_fork_source.rs [hunk 2]
```diff
@@ -91,6 +91,15 @@ impl ForkSource for HttpForkSource {
             .wrap_err("fork http client failed")
     }
 
+    fn get_transaction_details(
+        &self,
+        hash: H256,
+    ) -> eyre::Result<Option<zksync_types::api::TransactionDetails>> {
+        let client = self.create_client();
+        block_on(async move { client.get_transaction_details(hash).await })
```
Comment (MexicanAce) [inline src/http_fork_source.rs:99]: >We are not caching these responses due to their mutable status. We could cache when status == Finalized, but didn't seem worth it. wdyt?

Agreed, not worth it. We should add this comment here so future devs have the context as to why we chose not to cache the response

Comment (grw-ms) [reply src/http_fork_source.rs:102]: :+1: added comment

File: src/zks.rs [hunk 3]
```diff
@@ -197,12 +197,67 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> ZksNamespaceT
...
     {
-        not_implemented("zks_getTransactionDetails")
+        let inner = self.node.clone();
+        Box::pin(async move {
+            let reader = match inner.read() {
+                Ok(r) => r,
+                Err(_) => {
+                    return Err(into_jsrpc_error(Web3Error::InternalError));
+                }
+            };
```
Comment (nbaztec) [inline src/zks.rs:221]:
```suggestion
            let reader = match inner.read().map_err(|err| into_jsrpc_error(Web3Error::InternalError))?;
```

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**